File: examples/example-11/ruby/custom_param_step.rb
ParameterType(
    name: 'isoDate',
    regexp: /\d{4}-\d{2}-\d{2}/,
    type: Date,
    transformer: ->(date) { Date.parse(date) }
)

When('the user selects the date {isoDate}') do |date|
    # do something with the date...
end

File: examples/example-11/js/custom_param_step.js
const { defineParameterType, When } = require('@cucumber/cucumber');

defineParameterType({
    name: 'isoDate',
    regexp: /\d{4}-\d{2}-\d{2}/,
    transformer: d => new Date(d)
});

When('the user selects the date {isoDate}', async function(date) {
    // do something with the date...
});